pkgs/by-name/fr/friction-graphics/package.nix

```
@@ -0,0 +1,92 @@
+{ lib
+, clangStdenv
+, cmake
+, expat
+, fetchgit
+, ffmpeg_4-full
+, fontconfig
+, freetype
+, glib
+, gn
+, harfbuzzFull
+, icu
+, libGL
+, libjpeg_turbo
+, libpng
+, libsForQt5
+, libunwind
+, libwebp
+, ninja
+, pcre2
+, pkg-config
+, python3
+, qt5Full
+, xorg
+, zlib
+}:
+
+clangStdenv.mkDerivation rec {
+  pname = "friction.graphics";
+  version = "0.9.6";
+  src = fetchgit {
+    url = "https://github.com/friction2d/friction";
+    rev = version;
+    sha256 = "sha256-QjwkijsTAEEN1VqGtIfsNsHTM1DIRUp+hfABuXoj+nU=";
+    fetchSubmodules = true;
+  };
+  buildInputs = [
+    expat
+    ffmpeg_4-full
+    fontconfig
+    freetype
+    glib
+    gn
+    harfbuzzFull
+    icu
+    libGL
+    libjpeg_turbo
+    libpng
+    libsForQt5.qscintilla
+    libunwind
+    libwebp
+    ninja
+    pcre2
+    python3
+    qt5Full
+    xorg.libX11
+    zlib
+  ];
+  nativeBuildInputs = [ cmake pkg-config ];
+  cmakeFlags = [
+    "-DCMAKE_BUILD_TYPE=Release"
+    "-DCMAKE_INSTALL_PREFIX=bin"
+    "-DHARFBUZZ_INCLUDE_DIRS=${harfbuzzFull.dev}/include"
+    "-DQSCINTILLA_INCLUDE_DIRS=${libsForQt5.qscintilla}/include"
+    "-DQSCINTILLA_LIBRARIES_DIRS=${libsForQt5.qscintilla}/lib/"
+    "-DQSCINTILLA_LIBRARIES=libqscintilla2.so"
+    "-G Ninja"
+  ];
+  patchPhase = ''
+    sed -i 's|''${SKIA_SRC}/bin/gn)|${gn}/bin/gn)|' src/engine/CMakeLists.txt
+    sed -i 's|fontconfig|${fontconfig.lib}/lib/libfontconfig.so|' src/cmake/friction-common.cmake
+    grep -rl 'hb' | xargs sed -Ei 's/(["<])(hb.*\.h)/\1harfbuzz\/\2/'
+  '';
```

```suggestion

  postPatch = ''
    sed -i 's|''${SKIA_SRC}/bin/gn)|${gn}/bin/gn)|' src/engine/CMakeLists.txt
    sed -i 's|fontconfig|${fontconfig.lib}/lib/libfontconfig.so|' src/cmake/friction-common.cmake
    grep -rl 'hb' | xargs sed -Ei 's/(["<])(hb.*\.h)/\1harfbuzz\/\2/'
  '';

```
postPatch is usually placed right after src.
Also please use subtituteInPlace with --replace-fail if possbible so that we easily notice when things no longer match

---

I couldn't work out if the last one was possible to change to `substituteInPlace`. It's replacing all `#include <hb.h>` and `#include "hb.h"` to `#include <harfbuzz/hb.h>`, since it seems this is where the harfbuzz that's how to get the header in `harfbuzzFull`. I'm not sure if it's too bad if this doesn't loudly fail though — it would mean that `<hb.h>` is no longer being referenced.

I updated the first two however, thanks for the advice :+1: